### PR TITLE
Add linter for v2 golang template

### PR DESCRIPTION
**Pull request by IniZio**
ref #3666

### CODE REVIEW

File: devtools/gotemplatelinter/final-newline-rule.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package main
```
Comment (louischan-oursky) [inline devtools/gotemplatelinter/final-newline-rule.go:1]: We do not use hyphen in the filename of Go source files. Use underscore instead.

**Comment by tung2744** [comment]
Self implemented linter? :shocked:

File: devtools/gotemplatelinter/final-newline-rule.go (hunk 1 above)
Comment (IniZio) [reply devtools/gotemplatelinter/final-newline-rule.go]: Fixed up in 443f77e8186a454a369938ee5a8d6b3cdd0f6620

File: devtools/gotemplatelinter/indent_rule.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"strings"
+)
+
+type IndentationRule struct{}
+
+func (r IndentationRule) Check(content string, path string) []LintViolation {
+	var violations []LintViolation
+	lines := strings.Split(content, "\n")
+	for i, line := range lines {
+		if strings.HasPrefix(line, "\t") {
```
[the remark this answers is not in the document]
Comment (IniZio) [reply devtools/gotemplatelinter/indent_rule.go]: Fixed in [4f00f2c](https://github.com/authgear/authgear-server/pull/3710/commits/4f00f2c9c2ff2c0c530e19ab017dc4bc0f36a4e5)

File: devtools/gotemplatelinter/final_newline_rule.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"strings"
+)
+
+type FinalNewlineRule struct{}
+
+func (r FinalNewlineRule) Check(content string, path string) []LintViolation {
```
[the remark this answers is not in the document]
Comment (IniZio) [reply devtools/gotemplatelinter/final_newline_rule.go]: Fixed in [d7dc35b](https://github.com/authgear/authgear-server/pull/3710/commits/d7dc35b55ef2ee79ecbf8e041ce300ee094b611e)

**Comment by louischan-oursky** [comment]
@IniZio I added 3 commits.

**Pull request merged**